File: src/ee/ads/internal/MessageHelper.cpp
#include "ee/ads/internal/MessageHelper.hpp"

namespace ee {
namespace ads {
using Self = MessageHelper;

Self::MessageHelper(const std::string& prefix, const std::string& adId)
    : prefix_(prefix)
    , adId_(adId) {}

std::string Self::isLoaded() const {
    return prefix_ + "_isLoaded_" + adId_;
}

std::string Self::load() const {
    return prefix_ + "_load_" + adId_;
}

std::string Self::onLoaded() const {
    return prefix_ + "_onLoaded_" + adId_;
}

std::string Self::onFailedToLoad() const {
    return prefix_ + "_onFailedToLoad_" + adId_;
}

std::string Self::show() const {
    return prefix_ + "_show_" + adId_;
}

std::string Self::onFailedToShow() const {
    return prefix_ + "_onFailedToShow_" + adId_;
}

std::string Self::onClicked() const {
    return prefix_ + "_onClicked_" + adId_;
}

std::string Self::onClosed() const {
    return prefix_ + "_onClosed_" + adId_;
}

std::string Self::getPosition() const {
    return prefix_ + "_getPosition_" + adId_;
}

std::string Self::setPosition() const {
    return prefix_ + "_setPosition_" + adId_;
}

std::string Self::getSize() const {
    return prefix_ + "_getSize_" + adId_;
}

std::string Self::setSize() const {
    return prefix_ + "_setSize_" + adId_;
}

std::string Self::isVisible() const {
    return prefix_ + "_isVisible_" + adId_;
}

std::string Self::setVisible() const {
    return prefix_ + "_setVisible_" + adId_;
}

std::string Self::createInternalAd() const {
    return prefix_ + "_createInternalAd_" + adId_;
}

std::string Self::destroyInternalAd() const {
    return prefix_ + "_destroyInternalAd_" + adId_;
}
} // namespace ads
} // namespace ee

File: unity/Runtime/Ads/Internal/MessageHelper.cs
namespace EE.Internal {
    public class MessageHelper {
        private readonly string _prefix;
        private readonly string _adId;

        public MessageHelper(string prefix, string adId) {
            _prefix = prefix;
            _adId = adId;
        }

        public string IsLoaded => $"{_prefix}_isLoaded_{_adId}";
        public string Load => $"{_prefix}_load_{_adId}";
        public string OnLoaded => $"{_prefix}_onLoaded_{_adId}";
        public string OnFailedToLoad => $"{_prefix}_onFailedToLoad_{_adId}";
        public string Show => $"{_prefix}_show_{_adId}";
        public string OnFailedToShow => $"{_prefix}_onFailedToShow_{_adId}";
        public string OnClicked => $"{_prefix}_onClicked_{_adId}";
        public string OnClosed => $"{_prefix}_onClosed_{_adId}";

        public string GetPosition => $"{_prefix}_getPosition_{_adId}";
        public string SetPosition => $"{_prefix}_setPosition_{_adId}";
        public string GetSize => $"{_prefix}_getSize_{_adId}";
        public string SetSize => $"{_prefix}_setSize_{_adId}";
        public string IsVisible => $"{_prefix}_isVisible_{_adId}";
        public string SetVisible => $"{_prefix}_setVisible_{_adId}";

        public string CreateInternalAd => $"{_prefix}_createInternalAd_{_adId}";
        public string DestroyInternalAd => $"{_prefix}_destroyInternalAd_{_adId}";
    }
}